hack/update-helm-index.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash -e
+
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o pipefail
+
+this=`basename $0`
+
+usage () {
+cat << EOF
+Usage: $this [-h]
+
+Options:
+  --helm-repo-path      specify the path to the Helm repo (defaults to HELM_REPO_PATH)
+  --help/-h             show this help and exit
+
+Example:
+  - from a local path:
+	$this /path/to/nvidia-device-plugin-0.15.0.tgz
+  - from a URL:	
+    $this https://github.com/NVIDIA/k8s-device-plugin/archive/refs/tags/nvidia-device-plugin-0.15.0.tgz
+
+EOF
+}
+
+#
+# Parse command line
+#
+while [[ $# -gt 0 ]]; do
+	key="$1"
+	case $key in
+		--helm-repo-path)
+			HELM_REPO_PATH="$2"
+			shift 2
+			;;
+		--help/h) usage
+			exit 0
+			;;
+		*) usage
+			exit 1
+			;;
+	esac
+done
+
+# now we take the input from the user and check if is a path or url http/https
+asset_path="$@"
+
+if [ -z "$asset_path" ]; then
+	echo "No assets provided"
+	exit 1
+fi
+
+if [[ $asset_path =~ ^https?:// ]]; then
+	asset_url=$asset_path
+else
+	asset_local=$asset_path
+fi
+
+GH_REPO_PULL_URL="https://github.com/NVIDIA/k8s-device-plugin.git"
+git clone --local $$(pwd) $(HELM_REPO_PATH)
+git -C $(HELM_REPO_PATH) remote add github $(GH_REPO_PULL_URL)
+git -C $(HELM_REPO_PATH) fetch github gh-pages
+git -C $(HELM_REPO_PATH) checkout gh-pages
+
+pushd $HELM_REPO_PATH/stable > /dev/null
+# Download charts from release assets
+
+if [ -n "$asset_local" ]; then
+	echo "Copying $asset_local..."
+	cp -f $asset_local $HELM_REPO_PATH/stable
+fi
```

Comment:
I don't think this works the way we want. We are in `$HELM_REPO_PATH/stable` meaning that `$asset_local` would be interpreted relative to that path.

Reply:
`HELM_REPO_PATH` is the absolute path, so it doesn't matter where we are, the `cp` will perform the copy to the given path.

Reply:
What if we have to run these locally as in the "troubleshooting" case?